src/main/java/seedu/address/model/appointment/HospitalAppointmentList.java

```diff
@@ -50,4 +53,71 @@ public void bookAppointment(Appointment appointment) {
     public Iterator<Appointment> iterator() {
         return internalList.iterator();
     }
+
+    public void setAppointment(Appointment target, Appointment editedAppointment) {
+        requireAllNonNull(target, editedAppointment);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new AppointmentNotFoundException();
+        }
+
+        if (!target.equals(editedAppointment) && isADuplicateAppointment(editedAppointment)) {
+            throw new DuplicateAppointmentException();
+        }
+
+        internalList.set(index, editedAppointment);
+    }
+
+    public void setAppointments(HospitalAppointmentList replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
+
+    /**
+     * Replaces the contents of this list with {@code appointments}.
+     * {@code appointments} must not contain duplicate appointments.
+     */
+    public void setAppointments(List<Appointment> appointments) {
+        requireAllNonNull(appointments);
+        if (!appointmentsAreUnique(appointments)) {
+            throw new DuplicateAppointmentException();
+        }
+
+        internalList.setAll(appointments);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof HospitalAppointmentList // instanceof handles nulls
+                && internalList.equals(((HospitalAppointmentList) other).internalList));
+    }
+
+    @Override
+    public int hashCode() {
+        return internalList.hashCode();
+    }
+
+    /**
+     * Returns true if {@code appointments} contains only unique appointments.
+     */
+    private boolean appointmentsAreUnique(List<Appointment> appointments) {
+        for (int i = 0; i < appointments.size() - 1; i++) {
+            for (int j = i + 1; j < appointments.size(); j++) {
+                if (appointments.get(i).equals(appointments.get(j))) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
```

Comment:
Not a fan of this O(n^2) operation. 
I would have checked before insertion to make it O(n).
I think it might be possible to make it O(1) if you use HashSet and mess around with equals() and hashCode() but that's too risky for me.

Reply:
Despite this function only being used for 1 function, I like the SLAP being used.